doc/source/whatsnew/v0.24.2.rst

```diff
@@ -31,6 +31,7 @@ Fixed Regressions
 - Fixed regression in ``IntervalDtype`` construction where passing an incorrect string with 'Interval' as a prefix could result in a ``RecursionError``. (:issue:`25338`)
 - Fixed regression in :class:`Categorical`, where constructing it from a categorical ``Series`` and an explicit ``categories=`` that differed from that in the ``Series`` created an invalid object which could trigger segfaults. (:issue:`25318`)
 - Fixed pip installing from source into an environment without NumPy (:issue:`25193`)
+- Fixed name preserving bug of :meth:`DatetimeIndex.snap` (:issue:`25575`)
```

Comment:
move to 0.25.0

Bug in ....snap.....not preserving the `name` of the input ``Index``